### DISCUSSION TITLE
How to generate gradient text

### DISCUSSION THREAD

**Discussion by jishi92** [libvips/libvips#3637] [Q&A]
I use [vips_text](https://www.libvips.org/API/current/libvips-create.html#vips-text) () to make a text image.
And I want to generate a linear gradient text image, such as start at red and end at yellow.

I can't find any related params in the doc ,how should I do ?

**Comment by jcupitt** [comment]
Hello @jishi92,

You could make the text as a mask and then composite over a gradient image, or perhaps you could render the whole thing with SVG. It probably depends on your exact needs.

**Reply by jishi92** [reply]
hi @jcupitt ,
I don't have a gradient image. I want to generate a gradient text in a blank image.
such as:
![image](https://github.com/libvips/libvips/assets/22196414/c1ca8477-b0d1-4df3-a0f3-8cb68d8b51f0)

**Reply by jcupitt** [reply]
You can make a gradient image like this:

```ruby
#!/usr/bin/ruby

require 'vips'

def gradient(start, stop)
    lut = Vips::Image.identity / 255
    lut = lut * start + (lut * -1 + 1) * stop
    lut.colourspace(:srgb, :source_space => :lab)
end

def circular_index(width, height)
    # xy image with 0 at the centre
    xy = Vips::Image.xyz(width, height) - [width / 2, height / 2]

    # diagonal size
    size = (width ** 2 + height ** 2) ** 0.5

    # so 0 - 1 circular image, with 1 at the edge and 0 at the centre
    (xy[0] ** 2 + xy[1] ** 2) ** 0.5
end

red = [50, 100, 0]
green = [50, -100, 0]
rgb = circular_index(512, 512).maplut(gradient(red, green))

rgb.write_to_file("circ.png")
```

Generates:

![circ](https://github.com/libvips/libvips/assets/580843/3c004294-376c-4f73-9d5e-f6fc27a8cbae)

So you can make a gradient and composite it with some text.

**Reply by jcupitt** [reply]
Or with text as well:

```ruby
#!/usr/bin/ruby

require 'vips'

def gradient(start, stop)
    lut = Vips::Image.identity / 255
    lut = lut * start + (lut * -1 + 1) * stop
    lut.colourspace(:srgb, :source_space => :lab)
end

def circular_index(width, height)
    # xy image with 0 at the centre
    xy = Vips::Image.xyz(width, height) - [width / 2, height / 2]

    # diagonal size
    size = (width ** 2 + height ** 2) ** 0.5

    # so 0 - 1 circular image, with 1 at the edge and 0 at the centre
    (xy[0] ** 2 + xy[1] ** 2) ** 0.5
end

# make a mask with some text in the alpha
text = Vips::Image.text "hello world", dpi: 600
white = (Vips::Image.black(text.width, text.height) + [255, 255, 255]).copy(interpretation: :srgb)
mask = white.bandjoin text.invert

red = [50, 100, 0]
green = [50, -100, 0]
gradient = circular_index(mask.width, mask.height).maplut(gradient(red, green))

image = gradient.composite mask, "over"

image.write_to_file("result.png")
```

To make:

![result](https://github.com/libvips/libvips/assets/580843/730f0e18-8167-4d5b-bfee-b104d652ab14)
